client/src/components/MultiJurisdictionAudit/RoundManagement/OfflineBatchRoundDataEntry.tsx

```diff
@@ -0,0 +1,229 @@
+import React from 'react'
+import { useParams } from 'react-router-dom'
+import { Formik, FormikProps, Field, FieldArray } from 'formik'
+import { Button, HTMLTable } from '@blueprintjs/core'
+import useContestsJurisdictionAdmin from './useContestsJurisdictionAdmin'
+import { IRound } from '../useRoundsAuditAdmin'
+import useOfflineBatchResults, {
+  IOfflineBatchResults,
+} from './useOfflineBatchResults'
+
+interface IProps {
+  round: IRound
+}
+
+const OfflineBatchRoundDataEntry = ({ round }: IProps) => {
+  const { electionId, jurisdictionId } = useParams<{
+    electionId: string
+    jurisdictionId: string
+  }>()
+  const contests = useContestsJurisdictionAdmin(electionId, jurisdictionId)
+  const [batchResults, updateResults, finalizeResults] = useOfflineBatchResults(
+    electionId,
+    jurisdictionId,
+    round.id
+  )
+
+  if (!contests || !batchResults) return null
+
+  // We only support one contest for now
+  const contest = contests[0]
+
+  const { results, finalizedAt } = batchResults
+
+  return (
+    <Formik
+      initialValues={{ results }}
+      enableReinitialize
+      onSubmit={async values => {
+        // TODO validation to prevent partially empty rows
+        // Omit empty rows
+        const filteredResults = values.results.filter(
+          result =>
+            result.batchName !== '' &&
+            Object.values(result.choiceResults).every(
+              value => (value as string | number) !== ''
+            )
```

Comment:
@jonahkagan If you're wanting to omit empty rows, wouldn't it be better to test `.some(value => (value as string | number) !== '')`? That way you're just checking that at least one field has a value in it? Though if what you're wanting is to make sure they put in at least a '0' I think this would work, just that's not what's described in the comment, so I wanted to make sure.